Bumerang/app/src/main/java/com/seng480b/bumerang/fragments/RequestDetailFragment.java

```diff
@@ -97,28 +111,70 @@ public void setStyle(int style, @StyleRes int theme) {
         super.setStyle(style, theme);
     }
 
+    private boolean checkOfferValidity(int profileId, int requestId, boolean setErrorToast) {
+        if (isOwnRequest(profileId)) {
+            if (setErrorToast) {
+                Utility.longToast(getContext(), R.string.unable_to_accept_own_request);
+            }
+            return false;
+        } else if (alreadyAcceptedRequest(requestId)) {
+            if (setErrorToast) {
+                Utility.longToast(getContext(), R.string.unable_to_accept_request_again);
+            }
+            return false;
+        }
+        return true;
+    }
+
+    private boolean isOwnRequest(int profileId) {
+        return UserDataCache.getCurrentUser().getUserId() == profileId;
+    }
+
+    private boolean alreadyAcceptedRequest(int requestId) {
+        ArrayList<Offer> offers = UserDataCache.getOffers();
+        if (offers == null) {
+            return false;
+        }
+        for (Offer offer : offers) {
+            if (requestId == offer.getRequest().getRequestId()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
     @Override
     public void beforeAsyncTask() {
 
     }
 
     @Override
     public void afterAsyncTask(String result) {
-        String requester =  String.valueOf(UserDataCache.getRecentUser().getFirstName());
+        String requester = String.valueOf(UserDataCache.getRecentUser().getFirstName());
         if (result == null || result.equals("")) {
             longToast(getActivity(), R.string.error_message);
         } else {
-            String message;
-            if (request.getRequestType() == Request.RequestType.LEND ) {
-                message = String.format(Locale.getDefault(), getResources().getString(R.string.covered_you_message), requester);
-            } else {
-                message = String.format(Locale.getDefault(), getResources().getString(R.string.covered_them_message), requester);
-            }
+            // Create local copy of Offer
+            Offer offer = createOfferFromResult(result);
+            UserDataCache.addOffer(offer);
+            String message = String.format(Locale.getDefault(), getResources().getString(R.string.covered_them_message), requester);
```

Comment:
Sorry, I think something went wrong when I rebased here, don't remember deleting this haha. I'll change it to a switch anyway. Thanks for pointing it out!